src/interrupts/gyro_int.rs

```diff
@@ -0,0 +1,544 @@
+            /// Yaw (Z-axis) high event has occurred
+            zaxis_high_event: match reg_data & G_INT_Bitmasks::ZH_G {
+                0 => false,
+                _ => true,
+            },
+            /// Yaw (Z-axis) low event has occurred
+            zaxis_low_event: match reg_data & G_INT_Bitmasks::ZL_G {
+                0 => false,
+                _ => true,
+            },                
+        };
+        Ok(status)
+    }
+
+    /// Set gyroscope reference value for digital high-pass filter.
+    pub fn set_hipass_ref(&mut self, value: u8) -> Result<(), T::Error> {
+        self.interface.write(Sensor::Gyro, register::AG::REFERENCE_G.addr(), value)?;
+        Ok(())
+    }
+
+    /// Get gyroscope reference value for digital high-pass filter.
+    pub fn read_hipass_ref(&mut self) -> Result<u8, T::Error> {
+        let data: u8 = self.read_register(Sensor::Gyro, register::AG::REFERENCE_G.addr())?;
+        Ok(data)
+    }
+    
+    /// Gyroscope interrupt duration
+    /// Enable/disable wait function and define duration (for how many samples to wait before exiting interrupt)    
+    pub fn gyro_int_duration(&mut self, wait: FLAG, duration: u8) -> Result<(), T::Error> {
+        // read the current value of the register
+        
+        let mut reg_value = self.read_register(Sensor::Gyro, register::AG::INT_GEN_DUR_G.addr())?;
+
+        match wait {
+            FLAG::Enabled => reg_value & !0b1000_0000 | 0b1000_0000, // set bit
+            FLAG::Disabled => reg_value & !0b1000_0000, // clear bit
+        };
+
+        let duration: u8 = match duration { // clamp duration to 7 bit values
+            0..=127 => duration,
+            _ => 127,
+        };
+
+        reg_value &= !0b0111_1111; // clear the lowest 7 bits
+
+        reg_value |= duration; 
+
+
+        self.interface.write(Sensor::Gyro, register::AG::INT_GEN_DUR_G.addr(), reg_value)?;
+
+        Ok(())
+
+    }
+
+    /// Set gyroscope mode during inactivity, activation threshold and inactivity duration
+    pub fn set_gyro_inactivity(&mut self, setting: FLAG, threshold: u8, duration: u8) -> Result<(), T::Error> {
+        let mut data: u8 = 0;
+
+        match setting {
+            FLAG::Enabled => data | (1 << 7),
+            FLAG::Disabled => data,
+        };
+
+        match threshold {
+            0..=127 => data | threshold,
+            _ => data | 127,
+        };
+
+        self.interface.write(Sensor::Gyro, register::AG::ACT_THS.addr(), data)?;
+
+        self.interface.write(Sensor::Gyro, register::AG::ACT_DUR.addr(), duration)?;
+
+        Ok(())
+
+    }
+
+
+    // == COMPLETE THIS FUNCTION == 
+
+    /// Set threshold in ?
+    pub fn set_gyro_threshold(&mut self, 
+                                //threshold: f32
+                                x_ths: u16, y_ths: u16, z_ths: u16) -> Result<(), T::Error> {
+        
+        // let sensitivity = self.mag.scale.sensitivity();
+        // let mut data = threshold / sensitivity;
```

Comment:
We should decide if we want threshold, sensitivity, etc.

Reply:
OK, I cleaned up some of this `set_gyro_threshold` function, but it is not complete yet. I see now that I left it unfinished, I will look into that. Should I comment it out for now, the whole function?